src/veh_interact.cpp

```diff
@@ -1445,123 +1445,127 @@ void veh_interact::calc_overview()
+                    int pct = ( static_cast<double>( pt.ammo_remaining() ) / pt.ammo_capacity(
+                                    ammotype( "battery" ) ) ) * 100;
+                    int offset = 1;
+                    std::string fmtstring = "%i    %3i%%";
+                    if( pt.is_leaking() ) {
+                        fmtstring = "%i   " + leak_marker + "%3i%%" + leak_marker;
+                        offset = 0;
+                    }
+                    right_print( w, y, offset, item::find_type( pt.ammo_current() )->color,
+                                 string_format( fmtstring, pt.ammo_capacity( ammotype( "battery" ) ), pct ) );
+                };
+                overview_opts.emplace_back( "BATTERY", &vpr.part(), next_hotkey( vpr.part(), hotkey ), details );
             }
-        };
-        auto no_tank_details = []( const vehicle_part & pt, const catacurses::window & w, int y ) {
-            if( !pt.ammo_current().is_null() ) {
-                const itype *pt_ammo_cur = item::find_type( pt.ammo_current() );
-                double vol_L = to_liter( pt.ammo_remaining() * units::legacy_volume_factor /
-                                         pt_ammo_cur->stack_size );
-                int offset = 1;
-                std::string fmtstring = "%s  %5.1fL";
-                if( pt.is_leaking() ) {
-                    fmtstring = "%s  " + leak_marker + "%5.1fL" + leak_marker;
-                    offset = 0;
+        }
+
+        if( vpr.part().is_available() && ( vpr.part().is_reactor() || vpr.part().is_turret() ) ) {
+            auto details_ammo = []( const vehicle_part & pt, const catacurses::window & w, int y ) {
+                if( pt.ammo_remaining() ) {
+                    int offset = 1;
+                    std::string fmtstring = "%s   %5i";
+                    if( pt.is_leaking() ) {
+                        fmtstring = "%s  " + leak_marker + "%5i" + leak_marker;
+                        offset = 0;
+                    }
+                    right_print( w, y, offset, item::find_type( pt.ammo_current() )->color,
+                                 string_format( fmtstring, item::nname( pt.ammo_current() ), pt.ammo_remaining() ) );
                 }
-                right_print( w, y, offset, pt_ammo_cur->color,
-                             string_format( fmtstring, item::nname( pt.ammo_current() ),
-                                            round_up( vol_L, 1 ) ) );
+            };
+            if( vpr.part().is_reactor() && vpr.part().is_available() ) {
+                overview_opts.emplace_back( "REACTOR", &vpr.part(), next_hotkey( vpr.part(), hotkey ),
+                                            details_ammo );
+            }
+            if( vpr.part().is_turret() && vpr.part().is_available() ) {
+                overview_opts.emplace_back( "TURRET", &vpr.part(), next_hotkey( vpr.part(), hotkey ),
+                                            details_ammo );
```

Comment:
You have checked for `vpr.part().is_available()` in line 1526 above.

Reply:
i ve removed the is_avaliable() checks from the old if statements.
i probably can merge them into a single check, but will have to check if that doesn't interfere with an expansion to the ui that i am working on.